lib/extractor/index.ts

```diff
@@ -196,7 +196,12 @@ function layersWithLatestFileModifications(
 
       // not adding the original file to extractedLayers
       // and removing it from the set since it can be found in consecutive layers
-      if (removedFilesToIgnore.has(filename)) {
+      if (
+        removedFilesToIgnore.has(filename) ||
+        Array.from(removedFilesToIgnore).some((removedFile) =>
```

Comment:
That's exactly the type of edge cases I want brought up to feel comfortable with this change.

This is an edge case. In case both the folder and file are in the same directory this would happen.
Since `/app/abcEfg` start with `/app/abc`. I'll make a more strict comparison